src/ngraph/frontend/onnx_import/op/flatten.cpp

```diff
@@ -33,13 +33,10 @@ namespace ngraph
                     NodeVector inputs{node.get_ng_inputs()};
                     auto data = inputs.at(0);
                     auto axis = node.get_attribute_value<std::int64_t>("axis", 1);
+                    auto valid_axis =
```

Comment:
The name `valid_axis` is not a great name for `axis`. Please use reassign to `axis` instead throughout the PR.

Reply:
The axis attribute retrieved from onnx node is `int64_t`. The return type of `validate_axis()` is `size_t`. You have to cast from `size_t` to `int64_t` to avoid compilation warnings. While casting you can run into problems if a value held by `size_t` is larger than `int64_t` can fit though.

Reply:
Why is the return type of `validate_axis()` `size_t`?